mongodbatlas/data_source_mongodbatlas_database_user.go

```diff
@@ -81,21 +83,23 @@ func dataSourceMongoDBAtlasDatabaseUserRead(d *schema.ResourceData, meta interfa
 	if err := d.Set("username", dbUser.Username); err != nil {
 		return fmt.Errorf("error setting `username` for database user (%s): %s", d.Id(), err)
 	}
-	if err := d.Set("database_name", dbUser.DatabaseName); err != nil {
-		return fmt.Errorf("error setting `database_name` for database user (%s): %s", d.Id(), err)
+	if err := d.Set("auth_database_name", dbUser.DatabaseName); err != nil {
+		return fmt.Errorf("error setting `auth_database_name` for database user (%s): %s", d.Id(), err)
+	}
+	if err := d.Set("x509_type", dbUser.X509Type); err != nil {
+		return fmt.Errorf("error setting `x509_type` for database user (%s): %s", d.Id(), err)
 	}
 	if err := d.Set("roles", flattenRoles(dbUser.Roles)); err != nil {
 		return fmt.Errorf("error setting `roles` for database user (%s): %s", d.Id(), err)
 	}
-	log.Printf("LOG___ dbUser.Labels): %#+v\n", flattenLabels(dbUser.Labels))
 	if err := d.Set("labels", flattenLabels(dbUser.Labels)); err != nil {
 		return fmt.Errorf("error setting `labels` for database user (%s): %s", d.Id(), err)
 	}
 
 	d.SetId(encodeStateID(map[string]string{
-		"project_id":    projectID,
-		"username":      username,
-		"database_name": databaseName,
+		"project_id":         projectID,
+		"username":           username,
+		"auth_database_name": databaseName,
```

Comment:
I think this is a lot clearer as some found this confusing, thank you!  Is there any way to still support the old database_name as well without getting too messy?  If not then we'll need to be sure to log this as a breaking change in the Changelog, Fyi @marinsalinas

Reply:
I think that it is a breaking change, what do you think @marinsalinas ?

Reply:
Also the little change in the client breaked up all :S

Reply:
Oh no, if it's breaking too much we should just leave it as is then, sorry to suggest it as didn't think about how much impact it has and we are already short on time.  Let's just revert this and save the headache.

Reply:
It's okay it breaks a little bit  but with this PR all the breaking changes are fixed